flink-core/src/main/java/org/apache/flink/configuration/CheckpointingOptions.java

```diff
@@ -96,6 +96,16 @@
                                             "Recognized shortcut names are 'jobmanager' and 'filesystem'.")
                                     .build());
 
+    /** Whether to enable state change log. */
+    @Documentation.Section(value = Documentation.Sections.COMMON_STATE_BACKENDS)
+    @Documentation.ExcludeFromDocumentation("Hidden for now")
+    public static final ConfigOption<Boolean> ENABLE_STATE_CHANGE_LOG =
+            ConfigOptions.key("state.backend.enable-statechangelog")
```

Comment:
I think this option targets for checkpointing, and maybe `state.checkpoints.enable-changelog` looks better.

Reply:
To me, this parameter changes the behavior of backend in how it performs checkpointing. So I'd leave it under `state.backend` section. 

But probably `state.backend.changelog.enabled` would be better as there will be more settings (and it got a bit more votes in an offline vote:) )

Reply:
I agreed with what Roman said,

the state changelog is related to how the state is maintained and handled, and as a whole changing the behavior of checkpoint as well.

"state.backend.changelog.enabled" yes, this is better.